rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -100,38 +103,42 @@ def _paths_to(*file_names):
             # return the new path first.
             # chkstat prefers the new paths over the old ones, so callers that only care about the
             # first matching file must mimic that.
-            yield '/usr/share/permissions/' + name
-            yield '/etc/' + name
+            yield f'{SHARE_DIR}/{name}'
+            yield f'/etc/{name}'
 
     def check(self, pkg):
         if pkg.is_source:
             return
 
         permfiles = set()
-        # first pass, find and parse permissions.d files
+        # first pass, find and parse per-package drop-in files
         for f in pkg.files.keys():
-            for prefix in self._paths_to('permissions.d/'):
+            for prefix in itertools.chain(
```

Comment:
Why did you have to use this chain construction rather than add another yield in _paths_to?

Reply:
I mean, we are sourcing files from 3 well known directories. Do we need to generate their name at runtime?

Reply:
Maybe I'm blind but I seen no way to do that using the current interface of `_paths_to()`. It is being used for other kinds of path lookups. Passing `packages.d` as a parameter will yield an excess result.

Reply:
Yes, I realized after the first comment that you can't just reuse `_paths_to`. But in this case I think we can just do without the whole permissions dir name generation and use a static list

Reply:
I believe I have found a compromise. Instead of turning generators inside out, simply transform the output to a list and use regular concatenation. This should be clearer now.